src/extension/importer/core/src/main/java/org/geoserver/importer/Importer.java

```diff
@@ -1570,6 +1589,48 @@ public XStreamPersister initXStreamPersister(XStreamPersister xp) {
         return xp;
     }
 
+    /**
+     * Creates a style for the layer being imported from a resource that was included in the 
+     * directory or archive that the data is being imported from.
+     */
+    StyleInfo createStyleFromFile(Resource styleFile, ImportTask task) {
+        String ext = FilenameUtils.getExtension(styleFile.name());
+        if (ext != null) {
+            StyleHandler styleHandler = Styles.handler(ext);
+            if (styleHandler != null) {
+                try {
+                    StyledLayerDescriptor sld = styleHandler.parse(styleFile, null, null,
+                            new EntityResolverProvider(getGeoServer()).getEntityResolver());
+
+                    Style style = Styles.style(sld);
+                    if (style != null) {
+                        StyleInfo info = catalog.getFactory().createStyle();
+
+                        String styleName = styleGen.generateUniqueStyleName(task.getLayer().getResource());
+                        info.setName(styleName);
+                        
+                        info.setFilename(styleName + "." +ext);
+                        info.setFormat(styleHandler.getFormat());
+                        info.setWorkspace(task.getStore().getWorkspace());
+
+                        catalog.getResourcePool().writeStyle(info, styleFile.in());
```

Comment:
I believe the input stream will not get closed here, in a cursory look at the code in writeStyle and called methods I did not see the stream being closed.